fixing ordering of series in legend

ppisljar opened the pull request:
Summary: order of legend for series should be same as provided in the configuration.

Resolves #10543

thomasneirynck commented on src/ui/public/agg_response/tabify/_buckets.js:
```diff
@@ -34,5 +34,13 @@ export function AggResponseBucketsProvider() {
     }
   };
 
+  Buckets.prototype.reorder = function (params) {
```
The reorder call creates a side-effect, fairly significant, and is only called from the tabify module. I'm not a 100% sure if this would have other impacts on the code, but it does break encapsulation somewhat.

I'd consider making the `reorder()` method return a clone of the buckets object, but with the keys ordered against the filters. So in `tabify` you'd have something like:

```
 if (buckets.length) {
    const reorderedBuckets = buckets.orderBucketsAccordingToFilterParams(agg.params);
    //... and then just work with reorderedBuckets...
 }
```

Bargs commented on src/ui/public/agg_response/tabify/tabify.js:
```diff
@@ -37,13 +37,14 @@ export function AggResponseTabifyProvider(Private, Notifier) {
             });
           } else {
-            buckets.forEach(function (subBucket, key) {
+            reorderedBuckets.forEach(function (subBucket, key) {
```
Getting `reorderedBuckets is undefined` errors on this line because `orderBucketsAccordingToFilterParams` is still mutating `this` instead of returning a value.

ppisljar commented:
@thomas i moved back to the original approach (changing the Buckets itself)
Buckets is a collection and I think its ok if calling order on the collection does the ordering in place (same as js Array.sort) ... also the implementation gets simplified that way (buckets has a prototype forEach function which does some magic behind the scenes)

i also added support for range and ip range aggregations

ppisljar commented:
@bargs this should be ready for another look

Bargs commented:
Looks like they're passing now?

ppisljar commented:
yup, good for another look

alexfrancoeur commented:
@ppisljar I know we discussed but can't recall where we ended up, did we end up backporting to 5.x or is this fixed in 6.0?

Bargs commented:
PRs should have version labels on them
